build.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+
+set -eu
+
+is_substring()
+{
+    case "$2" in
+        *$1*)
+        return 0
+        ;;
+        *)
+        return 1
+        ;;
+    esac
+}
+
+OK=0
+CURRENT_DIR=$(pwd)
+MY_NAME=$(basename -- "$0")
+ROOTFS_ARCHIVE="rootfs.tar.xz"
+BUILD_DIRECTORY="${CURRENT_DIR}/build"
+ROOTFS_DIRECTORY="${BUILD_DIRECTORY}/rootfs"
+DEBIAN_VERSION="jessie"
+
+cleanup()
+{
+    if is_substring "${CURRENT_DIR}" "${BUILD_DIRECTORY}"
+    then
+        rm -rf "${BUILD_DIRECTORY}"
+    fi
+}
+
+prepare()
+{
+    if [ ! -d "${ROOTFS_DIRECTORY}/usr/bin" ]
+    then
+        mkdir -p "${ROOTFS_DIRECTORY}/usr/bin"
+    fi
+
+    if [ ! -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        cp /usr/bin/qemu-arm-static "${ROOTFS_DIRECTORY}/usr/bin/"
```

Comment:
I think this only works on directory's

Reply:
It can be done actually. Docker does this, and i tried (and failed it) after searching it, it is possible.
https://unix.stackexchange.com/questions/44373/single-bind-mounted-file-gets-out-of-sync-in-linux#44375

the trick is to do:
```sh
> mount --bind /usr/bin/qemu-arm-static /tmp/qemu-arm-static                                                                                                                                            mount: /tmp/qemu-arm-static: mount point does not exist.
> touch /tmp/qemu-arm-static                                                                                                                                                                                > sudo mount --bind /usr/bin/qemu-arm-static /tmp/qemu-arm-static                                                                                                                                           
-rw-r--r-- 1 oliver oliver 0 jul 24 12:30 /tmp/qemu-arm-static                                                                                                                                                                             > sudo umount /tmp/qemu-arm-static    
> rm /tmp/qemu-arm-static                                                                                                                                                                                                                                             
```
Can't decide which approach i find cleaner, as we still have to create a placeholder file, mount it, and remove the placeholder file. I really like how it works in docker (and docker does this transparently)
and since we cannot do this with a mkdir, but only with a placeholder file (touch) the code won't become much simpler?
thoughts?